File: solutions/swift/difference-of-squares/1/Sources/DifferenceOfSquares/DifferenceOfSquares.swift
import Foundation

class Squares {
  private let n: Int
  init (_ n: Int) {
    self.n = n
  }

  var sumOfSquares : Int {
    get {
      return (1...self.n).map {
      a in a*a
    }.reduce(0) {
      acc, a in acc + a
    }
    }
  }

  var squareOfSum : Int {
    get {
      let a = (1...self.n).reduce(0) {
      acc, a in acc + a
    }
    return a*a
    }
  }

  var differenceOfSquares : Int {
    get {
      return self.squareOfSum - self.sumOfSquares
    }
  }
}

File: solutions/c/difference-of-squares/2/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int sum_of_squares(unsigned int number) {
    unsigned int ans = 0;
    for (unsigned int i = 1; i <= number; i++) {
        ans += i*i;
    }
    return ans;
}
unsigned int square_of_sum(unsigned int number) {
    return (number + 1) * number * (number + 1) * number / 4;

}
unsigned int difference_of_squares(unsigned int number) {
    return square_of_sum(number) - sum_of_squares(number);
}